client/my-sites/checkout/composite-checkout/hooks/product-variants.tsx

```diff
@@ -81,7 +78,7 @@ export function useGetProductVariants(
 	debug( 'variantProductSlugs', variantProductSlugs );
 
 	const variantsWithPrices: AvailableProductVariant[] = useSelector( ( state ) => {
-		return computeProductsWithPrices( state, siteId, variantProductSlugs );
+		return siteId ? computeProductsWithPrices( state, siteId, variantProductSlugs ) : [];
```

Comment:
While loading or for siteless checkout, there is no site ID which could have caused this selector to fail.

Reply:
It turns out that this caused a regression. See https://github.com/Automattic/wp-calypso/pull/65005